cephfs/admin/fsadmin.go

```diff
@@ -117,6 +117,24 @@ func unmarshalResponseJSON(res []byte, status string, err error, v interface{})
 	return json.Unmarshal(res, v)
 }
 
+// extractPathResponse returns a cleaned up path from requests that get a path
+// unless an error is encountered, then an error is returned.
+func extractPathResponse(res []byte, status string, err error) (string, error) {
+	if err != nil {
+		return "", err
+	}
+	if status != "" {
+		return "", fmt.Errorf("error status: %s", status)
+	}
+	// if there's a trailing newline in the buffer strip it.
+	// ceph assumes a CLI wants the output of the buffer and there's
+	// no format=json mode available currently.
+	for len(res) >= 1 && res[len(res)-1] == '\n' {
+		res = res[:len(res)-1]
+	}
+	return string(res), nil
```

Comment:
Any reason not to do `strings.TrimSpaces(string(res))`, or use an other `strings.Trim*()` function?

Reply:
One part laziness, and one part a desire not to convert to a string until after we have removed the trailing newline. Plus I only wanted to remove newlines.